src/connect.js

```diff
@@ -169,11 +169,10 @@ function connect(mapStoreStateToConfig, mapDispatchToConfig, mergeConfig, option
 				var storeState = this.getStore().getState();
 				var prevStoreConfig = this.storeConfig_;
 				var storeConfig = this.getStoreConfig_(storeState);
-				if (object.shallowEqual(prevStoreConfig, storeConfig)) {
-					return;
-				}
-				this.hasStoreConfigChanged_ = true;
 				this.storeState = storeState;
```

Comment:
This will cause us to rerender, even when `pure` is true and the final config passed to the wrapped component didn't change. That will make `pure` stop working as expected.

I see the problem you're having, but I think the solution should be different. For now I believe a simple solution would be to pass `this.getStore().getState()` instead of `this.storeState` inside `getChildProps_`. This will keep `pure` working as expected and also make sure that we always pass the most recent state. I've checked and that's what `react-redux` does too. Do you think this makes sense?